beets/autotag/mb.py

```diff
@@ -429,6 +429,28 @@ def match_album(artist, album, tracks=None):
     if tracks is not None:
         criteria['tracks'] = six.text_type(tracks)
 
+    # Additional search cues from existing metadata.
+    if extra_tags is not None:
+        catalognum = extra_tags.get('catalognum', '').lower().strip()
+        if re.search(u'^[0-9a-z -]+$', catalognum):
+            criteria['catno'] = six.text_type(catalognum.replace(u' ', ''))
+
+        country = extra_tags.get('country', '').lower().strip()
+        if country:
+            criteria['country'] = six.text_type(country)
+
+        label = extra_tags.get('label', '').lower().strip()
+        if label:
+            criteria['label'] = six.text_type(label)
+
+        media = extra_tags.get('media', '').lower().strip()
+        if media:
+            criteria['format'] = six.text_type(media)
+
+        year = extra_tags.get('year', '')
+        if year:
+            criteria['date'] = six.text_type(year)
```

Comment:
Maybe this could be implemented with a loop (except for the special-case handling of `catalognum`)? Something like this:

```py
FIELDS_TO_KEYS = {
    'media': 'format',
    'country': 'country',
}
for field, key in FIELDS_TO_KEYS.items():
    value = extra_tags.get(key, '').lower().strip()
    if value:
        criteria[field] = six.text_type(value)
```